modules/sdk-coin-trx/src/lib/freezeBuilder.ts

```diff
@@ -0,0 +1,110 @@
+import { InvalidTransactionError, TransactionType } from '@bitgo/sdk-core';
+import { TransactionBuilder } from './transactionBuilder';
+import { Transaction } from './transaction';
+import { TransactionReceipt } from './iface';
+
+/**
+ * Valid resource types for Tron freezing
+ */
+export enum FreezeResource {
+  BANDWIDTH = 'BANDWIDTH',
+  ENERGY = 'ENERGY',
+}
+
+interface RawContract {
+  parameter: {
+    value: {
+      resource?: string;
+      frozen_balance?: number;
+      owner_address?: string;
+    };
+  };
+  type: string;
+}
+
+export class FreezeTransactionBuilder extends TransactionBuilder {
+  /** @inheritdoc */
+  protected get transactionType(): TransactionType {
+    return TransactionType.StakingActivate;
+  }
+
+  initBuilder(rawTransaction: TransactionReceipt | string): void {
+    this.transaction = this.fromImplementation(rawTransaction);
+    this.transaction.setTransactionType(this.transactionType);
+  }
+
+  validateTransaction(transaction: Transaction | TransactionReceipt): void {
+    if (transaction && typeof (transaction as Transaction).toJson === 'function') {
+      super.validateTransaction(transaction as Transaction);
+      const rawTx = (transaction as Transaction).toJson();
+      this.validateFreezeTransaction(rawTx);
+    } else {
+      this.validateFreezeTransaction(transaction as TransactionReceipt);
+    }
+  }
+
+  /**
+   * Validates if the transaction is a valid freeze transaction
+   * @param transaction The transaction to validate
```

Comment:
@param {TransactionReceipt} transaction - The transaction to validate

This is the format of jsdoc that should be followed in all places in this PR